projects/js-packages/charts/src/components/legend/use-chart-legend-data.ts

```diff
@@ -70,21 +71,34 @@ function processSeriesData(
 	renderGlyph?: React.ComponentType< unknown >
 ): LegendItemWithGlyph[] | LegendItemWithoutGlyph[] {
 	const mapper = ( series: SeriesData, index: number ) => {
+		const { stroke, lineStyles } = getSeriesStyles( series, index, theme );
 		const baseItem = createBaseLegendItem(
 			series.label,
 			showValues ? series.data?.length?.toString() || '0' : '',
-			theme.colors[ index % theme.colors.length ]
+			stroke
 		);
 
+		// Create shape style that includes comparison styling
+		const shapeStyle = {
```

Comment:
Should we extract shapeStyle as well?

Reply:
You mean add it to `getSeriesStyles`?

Reply:
Yes if you think it's a good idea too. Not a blocker either way.

Reply:
(or create a function like `getSeriesStyles` but for shapes)

Reply:
Done in [10150f6](https://github.com/Automattic/jetpack/pull/44676/commits/10150f6dd96cff259c5d96dda49b4c2ee7613172)